commerce_coordinator/apps/frontend_app_ecommerce/views.py

```diff
@@ -94,18 +94,23 @@ def get(self, request):
         if not request.user.lms_user_id:  # pragma: no cover
             raise PermissionDenied(detail="Could not detect LMS user id.")
 
-        order_data = OrderHistoryRequested.run_filter(request, params)
+        try:
+            order_data = OrderHistoryRequested.run_filter(request, params)
 
-        output_orders = []
+            output_orders = []
 
-        for order_set in order_data:
-            output_orders.extend(order_set['results'])
+            for order_set in order_data:
+                output_orders.extend(order_set['results'])
 
-        output = {
-            "count": request.query_params['page_size'],  # This suppresses the ecomm mfe Order History Pagination ctrl
-            "next": None,
-            "previous": None,
-            "results": sorted(output_orders, key=lambda item: date_conv(item["date_placed"]), reverse=True)
-        }
+            output = {
+                # This suppresses the ecomm mfe Order History Pagination control
+                "count": request.query_params['page_size'],
+                "next": None,
+                "previous": None,
+                "results": sorted(output_orders, key=lambda item: date_conv(item["date_placed"]), reverse=True)
+            }
 
-        return Response(output)
+            return Response(output)
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.error(f'UserOrdersView threw following exception: [{exc}]')
+            return Response(status=HTTP_400_BAD_REQUEST, data='Something went wrong!')
```

Comment:
[Needs Confirmation] 

We decided that we will throw overall error if any of the e-commerce system threw error. right? Like we wont show partial information from one of the system.

Reply:
I thought we send the partial order data 🤔

Reply:
yes, we won't show partial information from one system